src/compiler/scanner.ts

```diff
@@ -998,6 +1033,25 @@ module ts {
                     return token = SyntaxKind.EndOfFileToken;
                 }
                 let ch = text.charCodeAt(pos);
+
+                if (inJSXChild) {
+                    switch (ch) {
+                        case CharacterCodes.closeBrace:
+                            return pos++, token = SyntaxKind.CloseBraceToken;
+                        case CharacterCodes.openBrace:
+                            return pos++, token = SyntaxKind.OpenBraceToken;
+                        case CharacterCodes.lessThan:
+                            if (text.charCodeAt(pos + 1) === CharacterCodes.slash) {

```

Comment:
Interesting ambiguity with:    something </regex/.Something()...    Will have to consider the implications of this.